ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.test.KtLintAssertThat
+import com.pinterest.ktlint.test.LintViolation
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+
+class KtlintSuppressionRuleTest {
+    private val ktlintSuppressionRuleAssertThat =
+        KtLintAssertThat.assertThatRule { KtlintSuppressionRule() }
+
+    @Nested
+    inner class `Given a suppression annotation missing the rule set id prefix` {
+        @Test
+        fun `Given a @file Suppress annotation`() {
+            val code =
+                """
+                @file:Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")
```

Comment:
We can also test `@file:[Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar") SuppressWarnings("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")]`

Reply:
Few more variations
```
@file:[Suppress(names = arrayOf("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar"))]
@file:[Suppress(names = ["ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar"])]
@file:Suppress(names = arrayOf("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar"))
@file:Suppress(names = ["ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar"])
```

Reply:
I think you can add two more `@file:[Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar"), SomeOtherAnnotation]` or @file:[Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar"), SuppressWarnings]`

`@file:[AAA, Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")]` (here AAA is alphabetically sorted first annotation)

Reply:
Those tests won't add value once you know how the rule actually works. So from the unit test perspective those test won't increase the line coverage. The order of the annotations within the array is for example totally irrelevant for the logic. Also, it no longer makes a difference whether the annotation is included in a array or starts with the `@` when not included in an annotation array.